### PR TITLE
Update a few dependencies

### PR DESCRIPTION
**Pull request by schildbach**
Weirdly, BouncyCastle doesn't appear to be available on Maven Central for me, although IntelliJ alert that there is a new version, and according to bouncycastle.org it was released on Mar 31.

So I'm putting this to draft for now.

### CODE REVIEW

**Comment by msgilligan** [comment]
Yeah, it looks like Bouncy Castle 1.71 has not been released on Maven Central yet.

**Comment by msgilligan** [comment]
@schildbach A related question. Could we switch to the `bcprov-jdk18on` JAR?

From https://www.bouncycastle.org/latest_releases.html :

> **Java Version Details** With the arrival of Java 15. jdk15 is not quite as unambiguous as it was. The **jdk18on** jars are compiled to work with **anything** from Java 1.8 up. They are also multi-release jars so do support some features that were introduced in Java 9, Java 11, and Java 15. If you have issues with multi-release jars see the jdk15to18 release jars below.

> **Packaging Change (users of 1.70 or earlier)**: BC 1.71 changed the jdk15on jars to jdk18on so the base has now moved to Java 8. For earlier JVMs, or containers/applications that cannot cope with multi-release jars, you should now use the jdk15to18 jars.

File: core/build.gradle [hunk 1]
```diff
@@ -10,18 +10,18 @@ plugins {
 version = '0.17-SNAPSHOT'
 
 dependencies {
-    api 'org.bouncycastle:bcprov-jdk15to18:1.70'
-    api 'com.google.guava:guava:31.0.1-android'
+    api 'org.bouncycastle:bcprov-jdk15to18:1.71'
```
Comment (msgilligan) [inline core/build.gradle:13]: If you revert this to 1.70 the rest of the PR works.

**Review by msgilligan — CHANGES_REQUESTED** [review]

**Comment by schildbach** [comment]
I should have noted that I filed this upstream: https://github.com/bcgit/bc-java/issues/1139

File: core/build.gradle (hunk 1 above)
Comment (schildbach) [reply core/build.gradle:13]: If the missing BouncyCastle turns out to be more than a mistake, I will remove the BC update from this PR.

**Comment by schildbach** [comment]
Note it's still bundled, the jar can be downloaded here: https://www.bouncycastle.org/download/bcprov-jdk15to18-171.jar

**Comment by schildbach** [comment]
> A related question. Could we switch to the `bcprov-jdk18on` JAR?

Not without the guarantee that they don't use API that is not available in the Android API we want to support.

**Comment by schildbach** [comment]
Their release announcement which I just got via email says:

> Finally, for users of the maven repositories, 1.71 should appear in the
next few days.

**Comment by msgilligan** [comment]
> > A related question. Could we switch to the `bcprov-jdk18on` JAR?
> 
> Not without the guarantee that they don't use API that is not available in the Android API we want to support.

Right. Even if `bcprov-jdk18on` works with Android, there's no guarantee an updated version won't use an unsupported API.

**Comment by schildbach** [comment]
BC 1.71 is now on Maven Central, so I merged this.

**Pull request closed without merging**